tests/unit/Pipeline.cpp

```diff
@@ -0,0 +1,1227 @@
+/// \file Pipeline.cpp
+/// \brief Tests for Auto Pipe
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <algorithm>
+#include <memory>
+
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/IR/DerivedTypes.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/LegacyPassManager.h"
+#include "llvm/InitializePasses.h"
+#include "llvm/Pass.h"
+#include "llvm/Support/Casting.h"
+#include "llvm/Support/TargetSelect.h"
+#include "llvm/Support/YAMLTraits.h"
+#include "llvm/Transforms/Utils/Cloning.h"
+
+#include "revng/Pipeline/ContainerFactorySet.h"
+#include "revng/Pipeline/ContainerInspector.h"
+#include "revng/Pipeline/Context.h"
+#include "revng/Pipeline/Errors.h"
+#include "revng/Pipeline/InspectableLLVMContainer.h"
+#include "revng/Pipeline/LLVMContainerFactory.h"
+#include "revng/Pipeline/LLVMPipe.h"
+#include "revng/Pipeline/LibraryRegistry.h"
+#include "revng/Pipeline/Loader.h"
+#include "revng/Pipeline/Quantifier.h"
+#include "revng/Pipeline/Runner.h"
+#include "revng/Pipeline/Target.h"
+
+#define BOOST_TEST_MODULE Pipeline
+bool init_unit_test();
+#include "boost/test/unit_test.hpp"
+
+#include "revng/UnitTestHelpers/UnitTestHelpers.h"
+
+using namespace Pipeline;
+using namespace std;
+using namespace llvm;
+using KE = Exactness;
+
+static Granularity Root("Root");
+static Granularity FunctionGran("Function", Root);
+class RootKindType : public KindInspector<LLVMContainer> {
```

Comment:
What's the difference between `RootKind` and `RootKindType`?

Reply:
one it's just for tests and it was not handy to use the one referring to tagged functions